lib/src/disposable/disposable.dart

```diff
@@ -273,9 +280,13 @@ class Disposable implements _Disposable, DisposableManagerV3 {
     _internalDisposables.add(disposable);
 
     completer.future.catchError((e) {
-      _internalDisposables.remove(disposable);
+      if (!isDisposedOrDisposing) {
+        _internalDisposables.remove(disposable);
+      }
     }).then((_) {
-      _internalDisposables.remove(disposable);
+      if (!isDisposedOrDisposing) {
+        _internalDisposables.remove(disposable);
+      }
```

Comment:
Actually, I tried that just now and noticed that `whenComplete` re-raises exceptions, which we don't want in this case. Maybe there's a workaround.